app/uk/gov/hmrc/tai/connectors/CacheConnector.scala

```diff
@@ -85,18 +85,16 @@ class CacheConnector @Inject()(
   def find[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
     if (mongoConfig.mongoEncryptionEnabled) {
       val jsonDecryptor = new JsonDecryptor[T]()
-      taiCacheRepository.findById(cacheId.value) map {
-        case Some(cache) =>
+      OptionT(taiCacheRepository.findById(cacheId.value)).map {
```

Comment:
to refactor a bit your code u could extract a common method for taiCache and taiUpdate:

`private def findById[T](cacheId: CacheId, key: String)(f: String => Future[Option[CacheItem]])(implicit reads: Reads[T]): Future[Option[T]] =
    if (mongoConfig.mongoEncryptionEnabled) {
      val jsonDecryptor = new JsonDecryptor[T]()
      OptionT(f(cacheId.value)).map {
        cache =>
          (cache.data \ key).validateOpt[Protected[T]](jsonDecryptor).asOpt.flatten.map(_.decryptedValue)
      }.value.map(_.flatten)
    } recover {
      case JsResultException(_) => None
    } else {
      OptionT(f(cacheId.value)).map {
        cache => (cache.data \ key).validateOpt[T].asOpt.flatten
      }.value.map(_.flatten)
    }

  def find[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
    findById(cacheId, key)(taiCacheRepository.findById)

  def findUpdateIncome[T](cacheId: CacheId, key: String = defaultKey)(implicit reads: Reads[T]): Future[Option[T]] =
    findById(cacheId, key)(taiUpdateIncomeCacheRepository.findById)`